src/plugins/dashboard/public/dashboard_actions/filters_notification_popover.tsx

```diff
@@ -6,50 +6,66 @@
  * Side Public License, v 1.
  */
 
-import React, { useState } from 'react';
+import React, { useMemo, useState } from 'react';
 
 import {
   EuiButton,
-  EuiPopover,
-  EuiFlexItem,
-  EuiFlexGroup,
   EuiButtonIcon,
-  EuiPopoverTitle,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiPopover,
   EuiPopoverFooter,
+  EuiPopoverTitle,
 } from '@elastic/eui';
-import { EditPanelAction } from '@kbn/embeddable-plugin/public';
 
+import { css } from '@emotion/react';
+import { AggregateQuery, getAggregateQueryMode, isOfQueryType } from '@kbn/es-query';
+import { getEditPanelAction } from '@kbn/presentation-panel-plugin/public';
+import { FilterItems } from '@kbn/unified-search-plugin/public';
+import { FiltersNotificationActionApi } from './filters_notification_action';
 import { dashboardFilterNotificationActionStrings } from './_dashboard_actions_strings';
-import { FiltersNotificationActionContext } from './filters_notification_action';
-import { FiltersNotificationPopoverContents } from './filters_notification_popover_contents';
-
-export interface FiltersNotificationProps {
-  context: FiltersNotificationActionContext;
-  editPanelAction: EditPanelAction;
-  displayName: string;
-  icon: string;
-  id: string;
-}
 
-export function FiltersNotificationPopover({
-  editPanelAction,
-  displayName,
-  context,
-  icon,
-  id,
-}: FiltersNotificationProps) {
-  const { embeddable } = context;
+export function FiltersNotificationPopover({ api }: { api: FiltersNotificationActionApi }) {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
   const [disableEditbutton, setDisableEditButton] = useState(false);
 
+  const editPanelAction = getEditPanelAction();
+
+  const filters = useMemo(() => api.localFilters?.value, [api]);
+  const displayName = dashboardFilterNotificationActionStrings.getDisplayName();
+
+  const { queryString, queryLanguage } = useMemo(() => {
+    const localQuery = api.localQuery?.value;
+    if (!localQuery) return {};
+    if (isOfQueryType(localQuery)) {
+      if (typeof localQuery.query === 'string') {
+        return { queryString: localQuery.query };
+      } else {
+        return { queryString: JSON.stringify(localQuery.query, null, 2) };
+      }
+    } else {
+      setDisableEditButton(true);
+      const language: 'sql' | 'esql' | undefined = getAggregateQueryMode(localQuery);
+      return {
+        queryString: localQuery[language as keyof AggregateQuery],
+        queryLanguage: language,
+      };
+    }
+  }, [api, setDisableEditButton]);
```

Comment:
Great question, we'll definitely need to refactor this once local queries / filters are editable inline.

But for the time being, we can be certain that neither the API nor the `localQuery` will change reference while this popover is open. Really, the `api` is only there to satisfy the typescript rule exhaustive deps to avoid using an `// @ts-ignore`.

If we anticipated that the query would change over the lifetime of the flyout, we'd likely need to call `useLocalQuery` from `packages/presentation/presentation_publishing/interfaces/publishes_local_unified_search.ts` and then extract the string and language from there like:

```typescript
  const localQuery = useLocalQuery(api);
  const { queryString, queryLanguage } = useMemo(() => {
    if (!localQuery) return {};
    if (isOfQueryType(localQuery)) {
      if (typeof localQuery.query === 'string') {
        return { queryString: localQuery.query };
      } else {
        return { queryString: JSON.stringify(localQuery.query, null, 2) };
      }
    } else {
      setDisableEditButton(true);
      const language: 'sql' | 'esql' | undefined = getAggregateQueryMode(localQuery);
      return {
        queryString: localQuery[language as keyof AggregateQuery],
        queryLanguage: language,
      };
    }
  }, [localQuery, setDisableEditButton]);
```

Currently, this feels like overkill, because it would introduce an unnecessary subscription.